apps/uvc-app/uvc-gadget.cpp

```diff
@@ -284,7 +284,8 @@ void stopHandler(int code) { stop.store(true); }
  * V4L2 streaming related
  */
 
-static int v4l2_process_data(LocalDevice *device, uvc_device *udev) {
+static int v4l2_process_data(std::shared_ptr<LocalDevice> device,
```

Comment:
uvc_device *udev 
Astea de ce nu vor fi schimbate?

Reply:
The uvc gadget was initially implemented using C, the main purpose of this PR is to add smart pointers to the SDK api.